File: src/main/kotlin/at/fhooe/mc/wifipositioning/model/simulation/recording/ScannedAccessPoint.java
package at.fhooe.mc.wifipositioning.model.simulation.recording;


import at.fhooe.mc.wifipositioning.model.simulation.Position;
import com.fasterxml.jackson.annotation.JsonProperty;

public class ScannedAccessPoint implements Comparable<ScannedAccessPoint> {
    private String bssid;
    private String ssid;
    private int signalLevel;
    private int frequency;
    private String capabilities;
    private long timeStamp;
//    private Position position;
//    private int floor;

    public ScannedAccessPoint(String bssid, String ssid, int signalLevel, int frequency, String capabilities, long timeStamp, int floor) {
        this.bssid = bssid;
        this.ssid = ssid;
        this.signalLevel = signalLevel;
        this.frequency = frequency;
        this.capabilities = capabilities;
        this.timeStamp = timeStamp;
//        this.floor = floor;
    }

    public ScannedAccessPoint(){}

    public ScannedAccessPoint(String bssid, Position position, int floor) {
        this.bssid = bssid;
//        this.position = position;
//        this.floor = floor;
    }

    public ScannedAccessPoint(ScannedAccessPoint ap) {
        this.bssid = ap.bssid;
        this.ssid = ap.ssid;
        this.signalLevel = ap.signalLevel;
        this.frequency = ap.frequency;
        this.capabilities = ap.capabilities;
        this.timeStamp = ap.timeStamp;
//        this.position = new Position(ap.position.getX(), ap.position.getY());
//        this.floor = ap.floor;
    }


    public String getBssid() {
        return bssid;
    }

    @JsonProperty("bssid")
    public void setBssid(String bssid) {
        this.bssid = bssid;
    }

    public String getSsid() {
        return ssid;
    }

    @JsonProperty("ssid")
    public void setSsid(String ssid) {
        this.ssid = ssid;
    }

    public int getSignalLevel() {
        return signalLevel;
    }

    @JsonProperty("signalLevel")
    public void setSignalLevel(int signalLevel) {
        this.signalLevel = signalLevel;
    }

    public int getFrequency() {
        return frequency;
    }

    @JsonProperty("frequency")
    public void setFrequency(int frequency) {
        this.frequency = frequency;
    }

    public String getCapabilities() {
        return capabilities;
    }

    @JsonProperty("capabilities")
    public void setCapabilities(String capabilities) {
        this.capabilities = capabilities;
    }

    public long getTimeStamp() {
        return timeStamp;
    }

    @JsonProperty("timeStamp")
    public void setTimeStamp(long timeStamp) {
        this.timeStamp = timeStamp;
    }

    @Override
    public String toString(){
        return bssid;
    }

//    public Position getPosition() {
//        return position;
//    }
//
//    public void setPosition(Position position) {
//        this.position = position;
//    }
//
//    public int getFloor() {
//        return floor;
//    }
//
//    public void setFloor(int floor) {
//        this.floor = floor;
//    }

    @Override
    public int compareTo(ScannedAccessPoint o) {
        return Integer.compare(signalLevel,o.signalLevel);
    }
}

File: src/main/kotlin/at/fhooe/mc/wifipositioning/model/recording/ScannedAccessPoint.kt
package at.fhooe.mc.wifipositioning.model.recording

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Holds information about a scanned access point.
 *
 * @property bssid the BSSID address of the access point.
 * @property ssid the SSID of the access point
 * @property signalLevel the signal level (RSSI) of the Wi-Fi signal in dbM.
 * @property frequency the frequency of the Wi-Fi signal
 * @property capabilities the capabilities of the access point.
 * @property timestamp the timestamp of the detection.
 */
data class ScannedAccessPoint(@JsonProperty("bssid") val bssid: String,
                              @JsonProperty("ssid") val ssid: String,
                              @JsonProperty("signalLevel") var signalLevel: Int,
                              @JsonProperty("frequency") val frequency: Int,
                              @JsonProperty("capabilities") val capabilities: String,
                              @JsonProperty("timeStamp") val timestamp: Long): Comparable<ScannedAccessPoint> {

    override operator fun compareTo(o: ScannedAccessPoint): Int {
        return Integer.compare(signalLevel, o.signalLevel)
    }

    override fun toString(): String {
        return "$bssid [$signalLevel dbM]"
    }
}